src/page-modules/contact/travel-guarantee/travelGuaranteeFormMachine.ts

```diff
@@ -0,0 +1,366 @@
+import { TransportModeType } from '@atb-as/config-specs';
+import { assign, fromPromise, setup } from 'xstate';
+import { Line } from '../server/journey-planner/validators';
+import { machineEvents } from '../machineEvents';
+import { InputErrorMessages, formInputValidator } from '../formInputValidator';
+import { TranslatedString } from '@atb/translations';
+
+export const fetchMachine = setup({
+  types: {
+    context: {} as {
+      isIntialAgreementChecked: boolean;
+      isBankAccountForeign: boolean;
+      transportMode: TransportModeType | undefined;
+      line: Line | undefined;
+      fromStop: Line['quays'][0] | undefined;
+      toStop: Line['quays'][0] | undefined;
+      date: string;
+      plannedDepartureTime: string;
+      kilometersDriven: string;
+      reasonForTransportFailure:
+        | { id: string; name: TranslatedString }
+        | undefined;
+      feedback: string;
+      firstName: string;
+      lastName: string;
+      email: string;
+      address: string;
+      postalCode: string;
+      phoneNumber: string;
+      city: string;
+      bankAccountNumber: string;
+      IBAN: string;
+      SWIFT: string;
+      errorMessages: InputErrorMessages;
+    },
+    events: machineEvents,
+  },
+  guards: {
+    validateInputs: ({ context }) => formInputValidator(context),
+  },
+  actions: {
+    setIsIntialAgreementChecked: assign({
+      isIntialAgreementChecked: ({ context }) =>
+        !context.isIntialAgreementChecked,
+    }),
+    setIsBankAccountForeign: assign({
+      isBankAccountForeign: ({ context }) => !context.isBankAccountForeign,
+    }),
+    setTransportMode: assign({
+      transportMode: ({ event }) =>
+        (
+          event as {
+            type: 'SET_TRANSPORT_MODE';
+            transportMode: TransportModeType;
+          }
+        ).transportMode,
+    }),
+    setLine: assign({
+      line: ({ event }) => (event as { type: 'SET_LINE'; line: Line }).line,
+    }),
+    setFromStop: assign({
+      fromStop: ({ event }) =>
+        (
+          event as {
+            type: 'SET_FROM_STOP';
+            fromStop: Line['quays'][0];
+          }
+        ).fromStop,
+    }),
+    setToStop: assign({
+      toStop: ({ event }) =>
+        (
+          event as {
+            type: 'SET_TO_STOP';
+            toStop: Line['quays'][0];
+          }
+        ).toStop,
+    }),
+    setDate: assign({
+      date: ({ event }) =>
+        (
+          event as {
```

Comment:
Why do you need to use `as` here? 
Also, most of these "set" functions does the same, so should probably make an reusable function instead.